models/gif_psc_exp_multisynapse.cpp

```diff
@@ -0,0 +1,492 @@
+  }
+}
+
+void
+nest::gif_psc_exp_multisynapse::State_::get( DictionaryDatum& d,
+  const Parameters_& p ) const
+{
+  def< double >( d, names::V_m, y3_ );    // Membrane potential
+  def< double >( d, names::E_sfa, sfa_ ); // Adaptive threshold potential
+  def< double >( d, names::stc, stc_ );   // Spike-triggered current
+}
+
+void
+nest::gif_psc_exp_multisynapse::State_::set( const DictionaryDatum& d,
+  const Parameters_& p )
+{
+  updateValue< double >( d, names::V_m, y3_ );
+}
+
+nest::gif_psc_exp_multisynapse::Buffers_::Buffers_(
+  gif_psc_exp_multisynapse& n )
+  : logger_( n )
+{
+}
+
+nest::gif_psc_exp_multisynapse::Buffers_::Buffers_( const Buffers_&,
+  gif_psc_exp_multisynapse& n )
+  : logger_( n )
+{
+}
+
+/* ----------------------------------------------------------------
+ * Default and copy constructor for node
+ * ---------------------------------------------------------------- */
+
+nest::gif_psc_exp_multisynapse::gif_psc_exp_multisynapse()
+  : Archiving_Node()
+  , P_()
+  , S_()
+  , B_( *this )
+{
+  recordablesMap_.create();
+}
+
+nest::gif_psc_exp_multisynapse::gif_psc_exp_multisynapse(
+  const gif_psc_exp_multisynapse& n )
+  : Archiving_Node( n )
+  , P_( n.P_ )
+  , S_( n.S_ )
+  , B_( n.B_, *this )
+{
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::gif_psc_exp_multisynapse::init_state_( const Node& proto )
+{
+  const gif_psc_exp_multisynapse& pr =
+    downcast< gif_psc_exp_multisynapse >( proto );
+  S_ = pr.S_;
+}
+
+void
+nest::gif_psc_exp_multisynapse::init_buffers_()
+{
+  B_.spikes_.clear();   //!< includes resize
+  B_.currents_.clear(); //!< includes resize
+  B_.logger_.reset();   //!< includes resize
+  Archiving_Node::clear_history();
+}
+
+void
+nest::gif_psc_exp_multisynapse::calibrate()
+{
+  B_.logger_.init();
+
+  const double_t h = Time::get_resolution().get_ms();
+  V_.rng_ = kernel().rng_manager.get_rng( get_thread() );
+
+  const double_t tau_m = P_.c_m_ / P_.g_L_;
+
+  V_.P33_ = std::exp( -h / tau_m );
+  V_.P30_ = -1 / P_.c_m_ * numerics::expm1( -h / tau_m ) * tau_m;
+  V_.P31_ = -numerics::expm1( -h / tau_m );
+
+  V_.RefractoryCounts_ = Time( Time::ms( P_.t_ref_ ) ).get_steps();
+  // since t_ref_ >= 0, this can only fail in error
+  assert( V_.RefractoryCounts_ >= 0 );
+
+
+  // initializing adaptation (stc/sfa) variables
+  V_.P_sfa_.resize( P_.tau_sfa_.size(), 0.0 );
+  V_.P_stc_.resize( P_.tau_stc_.size(), 0.0 );
+
+  for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
+  {
+    V_.P_sfa_[ i ] = std::exp( -h / P_.tau_sfa_[ i ] );
+  }
+  S_.sfa_elems_.resize( P_.tau_sfa_.size(), 0.0 );
+
+  for ( size_t i = 0; i < P_.tau_stc_.size(); i++ )
+  {
+    V_.P_stc_[ i ] = std::exp( -h / P_.tau_stc_[ i ] );
+  }
+  S_.stc_elems_.resize( P_.tau_stc_.size(), 0.0 );
+
+  // initializing receptors
+  P_.receptor_types_.resize( P_.num_of_receptors_ );
```

Comment:
We actually don't need the `receptor_types_` array at all, just remove it. See details below in `handle(SpikeEvent&...)`.